projects/plugins/jetpack/tests/e2e/lib/setup-env.js

```diff
@@ -1,272 +0,0 @@
-/* eslint-disable jest/no-export */
-/**
- * External dependencies
- */
-import { get, wrap } from 'lodash';
-import fs from 'fs';
-/**
- * Internal dependencies
- */
-import { takeScreenshot } from './reporters/screenshot';
-import { logDebugLog, logHTML } from './page-helper';
-import logger from './logger';
-import { execWpCommand } from './utils-helper';
-import {
-	connectThroughWPAdmin,
-	loginToWpcomIfNeeded,
-	loginToWpSite,
-} from './flows/jetpack-connect';
-import TunnelManager from './tunnel-manager';
-import config from 'config';
-import path from 'path';
-
-const { E2E_TIMEOUT, E2E_DEBUG, CI, E2E_LOG_HTML } = process.env;
-let currentBlock;
-
-const defaultErrorHandler = async ( error, name ) => {
-	let filePath;
-
-	try {
-		filePath = await takeScreenshot( currentBlock, name );
-		reporter.addAttachment(
-			`Test failed: ${ currentBlock } :: ${ name }`,
-			fs.readFileSync( filePath ),
-			'image/png'
-		);
-	} catch ( e ) {
-		logger.warn( `Failed to add attachment to allure report: ${ e }` );
-	}
-
-	// If running tests in CI
-	if ( CI ) {
-		await logDebugLog();
-		logger.slack( {
-			type: 'failure',
-			message: { block: currentBlock, name, error },
-		} );
-		if ( filePath ) {
-			logger.slack( { type: 'file', message: filePath } );
-		}
-	}
-
-	if ( E2E_LOG_HTML ) {
-		logHTML();
-	}
-
-	if ( E2E_DEBUG ) {
-		console.log( error );
-		await jestPlaywright.debug();
```

Comment:
> couldn't find a good place for jestPlaywright.debug()

Can we take a similar approach? Put it into one of the event handlers?

Reply:
I personally almost never use it and don't want my tests to stop. I'm used to placing breakpoints in code. But I admit it can be useful., so I placed it under the `PAUSE_ON_FAILURE` env variable. This way, we can have both E2E_DEBUG without pause, with headful mode and everything else, and this pause on failure mode. Also, I used Playwright's new `page.pause()`, it looks more powerful.

Reply:
Oh interesting! I find it useful to be able to see the exact state of the site under test when failure occurs. Also useful to inspect locators